app/helpers/application_helper.rb

```diff
@@ -8,15 +8,18 @@ def bootstrap_class_for flash_type
     { success: "alert-success", errors: "alert-danger", alert: "alert-warning", notice: "alert-info" }[flash_type.to_sym] || flash_type.to_s
   end
 
-
+  def bootstrap_icon_for flash_type
+    { success: "ok", errors: "remove-circle", alert: "warning-sign", notice: "exclamation-sign" }[flash_type.to_sym] || "question-sign"
+  end
 
   def flash_messages(opts = {})
     flash.each do |msg_type, message|
-      concat(content_tag(:div, sanitize(message), class: "alert #{bootstrap_class_for(msg_type)} fade in") do
-              concat content_tag(:button, 'x', class: "close", data: { dismiss: 'alert' })
-              concat message
-            end)
-    end
+      concat(content_tag(:div, message, class: "alert #{bootstrap_class_for(msg_type)} fade in") do
+        concat content_tag(:button, 'x', class: "close", data: { dismiss: 'alert' })
+        concat content_tag(:i, nil, class: "glyphicon glyphicon-#{bootstrap_icon_for(msg_type)}")
+        concat message
+        end)
```

Comment:
`end` at 21, 8 is not aligned with `content_tag(:div, message, class: "alert #{bootstrap_class_for(msg_type)} fade in") do` at 17, 13 or `concat(content_tag(:div, message, class: "alert #{bootstrap_class_for(msg_type)} fade in") do` at 17, 6